docs/docs-content/getting-started/aws/deploy-manage-k8s-cluster-tf.md

```diff
@@ -0,0 +1,748 @@
+Palette for information about a specific pack, such as its unique ID, registry ID, available versions, and YAML values.
+
+Below is the data resource used to query Palette for information about the Kubernetes pack for version `1.29.0`.
+
+```hcl
+data "spectrocloud_pack" "aws_k8s" {
+  name    = "kubernetes"
+  version = "1.29.0"
+  registry_uid = data.spectrocloud_registry.public_registry.id
+}
+```
+
+Using the data resource helps you avoid manually entering the parameter values required by the cluster profile's
+`pack {}` block.
+
+#### Cluster
+
+The **clusters.tf** file contains the definitions required for deploying a host cluster to one of the infrastructure
+providers. To create an AWS host cluster, you must set the `deploy-aws` variable in the **terraform.tfvars** file to
+true.
+
+When deploying a cluster using Terraform, you must provide the same parameters as those available in the Palette UI for
+the cluster deployment step, such as the instance size and number of nodes. You can learn more about each parameter by
+reviewing the
+[AWS cluster resource](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/cluster_aws)
+documentation.
+
+```hcl
+resource "spectrocloud_cluster_aws" "aws-cluster" {
+  count = var.deploy-aws ? 1 : 0
+
+  name             = "aws-cluster"
+  tags             = concat(var.tags, ["env:aws"])
+  cloud_account_id = data.spectrocloud_cloudaccount_aws.account[0].id
+
+  cloud_config {
+    region       = var.aws-region
+    ssh_key_name = var.aws-key-pair-name
+  }
+
+  cluster_profile {
+    id = var.deploy-aws && var.deploy-aws-kubecost ? resource.spectrocloud_cluster_profile.aws-profile-kubecost[0].id : resource.spectrocloud_cluster_profile.aws-profile[0].id
+  }
+
+  machine_pool {
+    control_plane           = true
+    control_plane_as_worker = true
+    name                    = "control-plane-pool"
+    count                   = var.aws_control_plane_nodes.count
+    instance_type           = var.aws_control_plane_nodes.instance_type
+    disk_size_gb            = var.aws_control_plane_nodes.disk_size_gb
+    azs                     = var.aws_control_plane_nodes.availability_zones
+  }
+
+  machine_pool {
+    name          = "worker-pool"
+    count         = var.aws_worker_nodes.count
+    instance_type = var.aws_worker_nodes.instance_type
+    disk_size_gb  = var.aws_worker_nodes.disk_size_gb
+    azs           = var.aws_worker_nodes.availability_zones
+  }
+
+  timeouts {
+    create = "30m"
+    delete = "15m"
+  }
+}
+```
+
+## Terraform Tests
+
+Before starting the cluster deployment, test the Terraform code to ensure the resources will be provisioned correctly.
+Issue the following command in your terminal.
+
+```bash
+terraform test
+```
+
+The output should be similar to the following.
+
+```text hideClipboard
+Success! 16 passed, 0 failed.
+```
+
+## Input Variables
```

Comment:
Should this part be filled in before the user runs the tests?

Reply:
No, the tests provide the value of the variables inside the `variables` block. :)